### ISSUE TITLE
GridLayoutManager, spanSizeLookup, position is -1

### ISSUE THREAD

**Issue by kjsolo** [Fondesa/RecyclerViewDivider#17]
```kotlin
// position == -1
// MoreViewItem is last item
spanSizeLookup = object : GridLayoutManager.SpanSizeLookup() {
    override fun getSpanSize(position: Int): Int {
        val item = adapter.items[position] // ArrayIndexOutOfBoundsException
        if (item is MoreViewItem) {
            return 2
        }
        return 1
}

// notify insert
items.add(moreViewItem)
adapter.items = items
adapter.notifyItemInserted(items.lastIndex)
```

```
java.lang.ArrayIndexOutOfBoundsException: length=108; index=-1
        at java.util.ArrayList.get(ArrayList.java:439)
        at cn.soloho.xvideos.view.HomeFragment$onActivityCreated$$inlined$also$lambda$1.getSpanSize(HomeFragment.kt:57)
        at android.support.v7.widget.GridLayoutManager$SpanSizeLookup.getSpanGroupIndex(GridLayoutManager.java:971)
        at com.fondesa.recyclerviewdivider.extension.LayoutManagerExtensionsKt.getGroupIndex(LayoutManagerExtensions.kt:70)
        at com.fondesa.recyclerviewdivider.RecyclerViewDivider.getItemOffsets(RecyclerViewDivider.kt:104)
        at android.support.v7.widget.RecyclerView.getItemDecorInsetsForChild(RecyclerView.java:4792)
        at android.support.v7.widget.RecyclerView$LayoutManager.calculateItemDecorationsForChild(RecyclerView.java:9320)
        at android.support.v7.widget.GridLayoutManager.layoutChunk(GridLayoutManager.java:590)
        at android.support.v7.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1519)
        at android.support.v7.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:614)
        at android.support.v7.widget.GridLayoutManager.onLayoutChildren(GridLayoutManager.java:170)
        at android.support.v7.widget.RecyclerView.dispatchLayoutStep1(RecyclerView.java:3763)
        at android.support.v7.widget.RecyclerView.dispatchLayout(RecyclerView.java:3527)
        at android.support.v7.widget.RecyclerView.onLayout(RecyclerView.java:4082)
```

**Comment by Fondesa** [comment]
Fixed in [2.0.1](https://github.com/Fondesa/RecyclerViewDivider/releases/tag/2.0.1).
I contributed also to your repository updating the dependency and removing the `FIXME` lines.

**Comment by kjsolo** [comment]
Wow, Thank you
